src/lib/compose.js

```diff
@@ -51,34 +74,39 @@ export const some: Composer = validators => data =>
 
 // Helper
 type Flatten = (Array<Validator>, Array<Validator>) => Array<Validator>;
-const flatten: Flatten = (acc, vs) => [...acc, ...vs];
+export const flatten: Flatten = (acc, vs) => [...acc, ...vs];
 
 /**
  * Convert an object schema to an array of validators
  */
 type FromObjectSchema = ObjectSchema => Array<Validator>;
 export const fromObjectSchema: FromObjectSchema = (schema = {}) => {
-  const defaultVs = [validateIsObject];
-
-  if (!isObject(schema)) return defaultVs;
-
-  return Object.keys(schema)
-    .map(k => {
-      const rules = schema[k];
-
-      return Object.keys(rules)
-        .map(r => {
-          if (r === 'required' && rules[r]) {
-            return rules[r] === true ? [validateObjHasKey(k)] : [];
-          } else if (r === 'type' && rules[r]) {
-            return [validateObjPropHasType(rules[r])(k)];
-          } else if (r === 'validator' && rules[r]) {
-            return [validateObjPropPasses(rules[r])(k)];
-          } else return [];
-        })
-        .reduce(flatten, []);
-    })
-    .reduce(flatten, defaultVs);
+  if (!isObject(schema))
+    return processSchemaError(err(['Schemas must be objects']));
+
+  const schemaResult = validateAsObjectSchema(schema);
+  if (isErr(schemaResult)) return processSchemaError(schemaResult);
+
+  const requiredChecks = Object.keys(schema).map(
+    k => (schema[k].required ? validateObjHasKey(k) : alwaysOk())
+  );
```

Comment:
Actually appears this is no longer required, most likely because of structural changes made elsewhere - my next push will have this removed so we can confirm, but running `$ yarn flow` locally returns no errors